src/decode.h

```diff
@@ -778,58 +768,33 @@ void CaptureStatsSetup(ThreadVars *tv, CaptureStats *s);
         }                                           \
     } while(0)
 
-/* macro's for setting the action
- * handle the case of a root packet
- * for tunnels */
-
-#define PACKET_SET_ACTION(p, a) (p)->action = (a)
-
-static inline void PacketSetAction(Packet *p, const uint8_t a)
+static inline void PacketDrop(Packet *p, const uint8_t action, enum PacketDropReason r)
 {
-    if (likely(p->root == NULL)) {
-        PACKET_SET_ACTION(p, a);
-    } else {
-        PACKET_SET_ACTION(p->root, a);
+    if (p->drop_reason == PKT_DROP_REASON_NOT_SET)
+        p->drop_reason = (uint8_t)r;
+    if (p->root) {
+        p->root->action |= (action & ACTION_DROP_REJECT);
```

Comment:
remove mask, we never set action flags other than drop/reject anymore (debug validation can check that)